salt/states/cmd.py

```diff
@@ -917,8 +928,9 @@ def run(name,
 
     # Wow, we passed the test, run this sucker!
     try:
-        cmd_all = __salt__['cmd.run_all'](
-            name, timeout=timeout, python_shell=True, **cmd_kwargs
+        run_cmd = 'cmd.run_all' if not root else 'cmd.run_chroot'
+        cmd_all = __salt__[run_cmd](
```

Comment:
Given that we are calling into a few other modules if we chroot here, we may need to expand our exception handling.

Reply:
Right.

Reply:
I change it to Exception, not sure if is too generic.